<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>300</width>
    <height>300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="statusTip">
   <string>Left: Rotate, Middle: Pan, Right/Scroll: Zoom, F: Reset Camera</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>240</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::VLine</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>248</width>
           <height>525</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QGroupBox" name="emitter_groupBox">
              <property name="title">
               <string>Settings</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <property name="spacing">
                <number>3</number>
               </property>
               <property name="leftMargin">
                <number>3</number>
               </property>
               <property name="rightMargin">
                <number>3</number>
               </property>
               <item>
                <layout class="QFormLayout" name="formLayout">
                 <property name="fieldGrowthPolicy">
                  <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
                 </property>
                 <property name="horizontalSpacing">
                  <number>3</number>
                 </property>
                 <property name="verticalSpacing">
                  <number>6</number>
                 </property>
                 <item row="0" column="0">
                  <widget class="QLabel" name="lblTileRadius">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>Patch Tile Radius</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QSpinBox" name="patchTileRadius"/>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="lblPatchSize">
                   <property name="text">
                    <string>Patch Size (Vertices)</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QSpinBox" name="patchSize">
                   <property name="maximum">
                    <number>500</number>
                   </property>
                   <property name="singleStep">
                    <number>2</number>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="lblPatchSpacing">
                   <property name="text">
                    <string>Patch Spacing</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QDoubleSpinBox" name="patchSpacing"/>
                 </item>
                 <item row="3" column="0" colspan="2">
                  <widget class="Line" name="line_2">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="0">
                  <widget class="QLabel" name="lblMaxHeight">
                   <property name="text">
                    <string>Max Height</string>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="1">
                  <widget class="QSpinBox" name="maxHeight">
                   <property name="maximum">
                    <number>10000</number>
                   </property>
                   <property name="singleStep">
                    <number>500</number>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="0">
                  <widget class="QLabel" name="lblHeightMapScale">
                   <property name="text">
                    <string>HeightMap Repeat</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <widget class="QDoubleSpinBox" name="heightMapScale">
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="6" column="0">
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>HeightMap Offset</string>
                   </property>
                  </widget>
                 </item>
                 <item row="6" column="1">
                  <layout class="QHBoxLayout" name="horizontalLayout_2">
                   <property name="spacing">
                    <number>0</number>
                   </property>
                   <item>
                    <widget class="QDoubleSpinBox" name="heightMapOffsetX">
                     <property name="maximum">
                      <double>2048.000000000000000</double>
                     </property>
                     <property name="singleStep">
                      <double>10.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QDoubleSpinBox" name="heightMapOffsetY">
                     <property name="maximum">
                      <double>2048.000000000000000</double>
                     </property>
                     <property name="singleStep">
                      <double>10.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="7" column="0" colspan="2">
                  <widget class="Line" name="line">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
                 <item row="8" column="0">
                  <widget class="QLabel" name="lblNoiseMapScale">
                   <property name="text">
                    <string>NoiseMap Repeat</string>
                   </property>
                  </widget>
                 </item>
                 <item row="8" column="1">
                  <widget class="QDoubleSpinBox" name="noiseMapScale"/>
                 </item>
                 <item row="9" column="0">
                  <widget class="QLabel" name="lblNoiseMapWeight">
                   <property name="text">
                    <string>NoiseMap Weight</string>
                   </property>
                  </widget>
                 </item>
                 <item row="9" column="1">
                  <widget class="QDoubleSpinBox" name="noiseMapWeight">
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="10" column="0" colspan="2">
                  <widget class="Line" name="line_3">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
                 <item row="11" column="0">
                  <widget class="QLabel" name="lblWireframe">
                   <property name="text">
                    <string>Wireframe</string>
                   </property>
                  </widget>
                 </item>
                 <item row="11" column="1">
                  <widget class="QCheckBox" name="wireframe"/>
                 </item>
                 <item row="12" column="0">
                  <widget class="QLabel" name="label_4">
                   <property name="text">
                    <string>Textured</string>
                   </property>
                  </widget>
                 </item>
                 <item row="12" column="1">
                  <widget class="QCheckBox" name="textured">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="13" column="0">
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Texture Repeat</string>
                   </property>
                  </widget>
                 </item>
                 <item row="13" column="1">
                  <widget class="QDoubleSpinBox" name="textureScale"/>
                 </item>
                 <item row="14" column="0" colspan="2">
                  <widget class="Line" name="line_smoothing">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
                 <item row="15" column="0">
                  <widget class="QLabel" name="lblSmoothingIterations">
                   <property name="enabled">
                    <bool>true</bool>
                   </property>
                   <property name="text">
                    <string>Smoothing Iterations</string>
                   </property>
                  </widget>
                 </item>
                 <item row="15" column="1">
                  <widget class="QSpinBox" name="smoothingIterations"/>
                 </item>
                 <item row="16" column="0">
                  <widget class="QLabel" name="lblSmoothingWeight">
                   <property name="text">
                    <string>Smoothing Weight</string>
                   </property>
                  </widget>
                 </item>
                 <item row="16" column="1">
                  <widget class="QDoubleSpinBox" name="smoothingWeight">
                   <property name="maximum">
                    <double>1.000000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.100000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="17" column="0">
                  <widget class="QLabel" name="lblSmoothingRadius">
                   <property name="text">
                    <string>Smoothing Radius</string>
                   </property>
                  </widget>
                 </item>
                 <item row="17" column="1">
                  <widget class="QSpinBox" name="smoothingRadius"/>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="GLWidget" name="glView" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
